versioneer.py

```diff
@@ -1647,6 +1647,25 @@ def make_release_tree(self, base_dir, files):
                                   self._versioneer_generated_versions)
     cmds["sdist"] = cmd_sdist
 
+    import distutils.command.clean
+    import shutil
+
+    class cmd_clean(distutils.command.clean.clean):
+        def run(self):
+            import glob
+            with open('.clean', 'r') as f:
+                ignores = f.read()
+                for wildcard in filter(bool, ignores.split('\n')):
+                    for filename in glob.glob(wildcard):
+                        try:
+                            os.remove(filename)
+                        except OSError:
+                            shutil.rmtree(filename, ignore_errors=True)
+
+            # It's an old-style class in Python 2.7...
+            distutils.command.clean.clean.run(self)
+    cmds["clean"] = cmd_clean
+
```

Comment:
Don't put that in versioneer, it's going to the lost the next time we update it.

Reply:
The thing is that the setup call at the end of `setup.py` uses cmds list from versioneer.

Reply:
You can get the cmdclass from versioneer, add the custom clean command and then pass it to setup().